### PR TITLE
feat: support sequence return

### PR DESCRIPTION
**Pull request by Sychorius**
Change-Id: I95036796bdffd559cb713ec3485fa6f064eaf1fa

#### What type of PR is this?
feat: Support sequence returns with one single `Return`

#### What this PR does / why we need it (en: English/zh: Chinese):

en: Support sequence returns with one single `Return`
zh: 支持在一个return里序列化设置顺序返回值

### CODE REVIEW

File: mock_sequence.go
Comment (lsqlebai) [inline mock_sequence.go:1]:
```
func (s *sequence) Then(value ...interface{}) SequenceOpt {
	s.values = append(s.values, value)
	return s
}
```
fn Then() and Build()
there need more check with times。then slice < times or slice > times。
or donot have times?

File: mock_sequence.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mockey
+
+import "github.com/bytedance/mockey/internal/tool"
+
+type SequenceOpt interface {
```
Comment (ycydsxy) [inline mock_sequence.go:21]: why `SequeceOpt` and `sequenceOpt` are all interfaces?

**Review by ycydsxy — APPROVED** [review]

File: mock_sequence.go (hunk 1 above)
Comment (Sychorius) [reply mock_sequence.go:21]: Provide an interface that users can implement it by themselves

File: mock_sequence.go
Comment (Sychorius) [reply mock_sequence.go:1]: Updated, now a `sequence` has its own `curV` and `curT`, and `sequenceValue` has its own repeat-times config `t`.
```go
type sequence struct {
	MockeyPrivate // make sure it does implements mockey SequenceOpt
	values        []*sequenceValue
	curV          int // current value
	curT          int // current value times
}

type sequenceValue struct {
	v []interface{}
	t int
}
```

**Pull request merged**